File: mobile/src/main/java/org/openhab/habdroid/ui/widget/HelpIconShowingPreferenceUtil.java
package org.openhab.habdroid.ui.widget;

import android.content.Context;
import android.content.Intent;
import android.net.Uri;
import android.view.View;
import android.widget.ImageView;
import androidx.appcompat.widget.TooltipCompat;

public class HelpIconShowingPreferenceUtil {
    public static void setupHelpIcon(Context context, ImageView helpIcon,
            String url, String contentDescription) {
        final Uri howToUri = Uri.parse(url);
        final Intent intent = new Intent(Intent.ACTION_VIEW, howToUri);

        if (intent.resolveActivity(context.getPackageManager()) != null) {
            helpIcon.setOnClickListener(v -> context.startActivity(intent));
            helpIcon.setContentDescription(contentDescription);
            TooltipCompat.setTooltipText(helpIcon, contentDescription);
        } else {
            helpIcon.setVisibility(View.GONE);
        }
    }

    public static void updateHelpIconAlpha(ImageView helpIcon, boolean isEnabled) {
        if (helpIcon != null) {
            helpIcon.setAlpha(isEnabled ? 1.0f : 0.5f);
        }
    }
}

File: mobile/src/main/java/org/openhab/habdroid/ui/widget/HelpIconShowingPreferenceUtil.kt
package org.openhab.habdroid.ui.widget

import android.content.Context
import android.content.Intent
import android.net.Uri
import android.view.View
import android.widget.ImageView
import androidx.appcompat.widget.TooltipCompat

object HelpIconShowingPreferenceUtil {
    fun setupHelpIcon(context: Context, helpIcon: ImageView,
                      url: String, contentDescription: String) {
        val howToUri = Uri.parse(url)
        val intent = Intent(Intent.ACTION_VIEW, howToUri)

        if (intent.resolveActivity(context.packageManager) != null) {
            helpIcon.setOnClickListener { v -> context.startActivity(intent) }
            helpIcon.contentDescription = contentDescription
            TooltipCompat.setTooltipText(helpIcon, contentDescription)
        } else {
            helpIcon.visibility = View.GONE
        }
    }

    fun updateHelpIconAlpha(helpIcon: ImageView?, isEnabled: Boolean) {
        if (helpIcon != null) {
            helpIcon.alpha = if (isEnabled) 1.0f else 0.5f
        }
    }
}
